From a04689ee7a2600a1466354096123c57ccd1e1dc7 Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Wed, 7 Oct 2020 15:24:04 -0300
Subject: [PATCH 163/267] nptl: Add missing cancellation flags on
 futex_internal and pselect32

It fixes the tst-cancelx{4,5} and tst-cancel24-{static} regression on
some platforms (arm and sparc32).

Checked on arm-linux-gnueabihf and sparcv9-linux-gnu.
---
 sysdeps/nptl/Makefile            | 1 +
 sysdeps/unix/sysv/linux/Makefile | 1 +
 2 files changed, 2 insertions(+)

diff --git a/sysdeps/nptl/Makefile b/sysdeps/nptl/Makefile
index a65be3b7ea..8dfb6fdf1e 100644
--- a/sysdeps/nptl/Makefile
+++ b/sysdeps/nptl/Makefile
@@ -18,6 +18,7 @@
 
 ifeq ($(subdir),nptl)
 libpthread-sysdep_routines += errno-loc futex-internal
+CFLAGS-futex-internal.c += -fexceptions -fasynchronous-unwind-tables
 endif
 
 ifeq ($(subdir),rt)
diff --git a/sysdeps/unix/sysv/linux/Makefile b/sysdeps/unix/sysv/linux/Makefile
index a54eb75d74..de9e86553b 100644
--- a/sysdeps/unix/sysv/linux/Makefile
+++ b/sysdeps/unix/sysv/linux/Makefile
@@ -69,6 +69,7 @@ CFLAGS-vmsplice.c = -fexceptions -fasynchronous-unwind-tables
 CFLAGS-splice.c = -fexceptions -fasynchronous-unwind-tables
 CFLAGS-open_by_handle_at.c = -fexceptions -fasynchronous-unwind-tables
 CFLAGS-sync_file_range.c = -fexceptions -fasynchronous-unwind-tables
+CFLAGS-pselect32.c = -fexceptions -fasynchronous-unwind-tables
 CFLAGS-tst-writev.c += "-DARTIFICIAL_LIMIT=(0x80000000-sysconf(_SC_PAGESIZE))"
 
 sysdep_headers += sys/mount.h sys/acct.h \
-- 
2.29.0

